2017/ap/problems/mashup/mashup.adoc

```diff
@@ -0,0 +1,556 @@
+
+[source,sql]
+----
+q = request.args.get("q") + "%"
+db.execute("SELECT * FROM places WHERE postal_code LIKE :q", q=q)
+----
+
+Notice how this example appends `%` to the user's input, which happens to be SQL's "wildcard" character that means "match any number of characters." The effect is that this query will return rows whose postal codes match whatever the user typed followed by any number of other characters. In other words, any of `0`, `02`, `021`, `0213`, and `02138` might return rows, as might any of `0`, `06`, `065`, `0651`, and `06511`.
+
+If you'd like to support searching by more than just postal codes, keep in mind that SQL supports `OR` and `AND`!
+
+To test `/search`, even before your text box is operational, simply visit URLs like
+
+* `pass:[https://ide50-username.cs50.io/search?q=02138]`
+* `pass:[https://ide50-username.cs50.io/search?q=Cambridge+MA]`
+* `pass:[https://ide50-username.cs50.io/search?q=Cambridge,+MA]`
+* `pass:[https://ide50-username.cs50.io/search?q=Cambridge,+Massachusetts]`
+* `pass:[https://ide50-username.cs50.io/search?q=Cambridge,+Massachusetts,+US]`
+
+or
+
+* `pass:[https://ide50-username.cs50.io/search?q=06511]`
+* `pass:[https://ide50-username.cs50.io/search?q=New+Haven+CT]`
+* `pass:[https://ide50-username.cs50.io/search?q=New+Haven,+CT]`
+* `pass:[https://ide50-username.cs50.io/search?q=New+Haven,+Connecticut]`
+* `pass:[https://ide50-username.cs50.io/search?q=New+Haven,+Connecticut,+US]`
+
+and other such variants, where `username` is your own username, to see if you get back the JSON you expect. Again, though, we leave it to you to decide just how supportive `/search` will be of such variants. The more flexible, though, the better! Try to implement features that you yourself would expect as a user!
+
+Feel free to tinker with the staff's solution at http://pset8.cs50.net/, inspecting its HTTP requests via Chrome's Network tab as needed, if unsure how your own code should work!
+
+==== `scripts.js`
+
+First, toward the top of `scripts.js`, you'll see an anonymous function, inside of which is a definition of `options`, an object, one of whose keys is `center`, the value of which is an object with two keys of its own, `lat`, and `lng`. Per the comment alongside that object, your mashup's map is currently centered on Stanford, California. (D'oh.) Change the coordinates of your map's center to Cambridge (42.3770, -71.1256) or New Haven (41.3184, -72.9318) or anywhere else! (Though be sure to choose coordinates in the US if you downloaded `US.txt`!) Once you save your changes and reload your map, you should find yourself there! Zoom out as needed to confirm visually.
+
+As before, feel free to tinker with the staff's solution at http://pset8.cs50.net/, inspecting its HTTP requests via Chrome's Network tab as needed, if unsure how your own code should work!
```

Comment:
Replace with `Feel free to tinker with the staff's solution at http://mashup.cs50.net/, inspecting its HTTP requests via Chrome's Network tab as needed, if unsure how your own code should work!`